ko

toka13902 opened the pull request.

toka13902 committed:
ko

sshwang commented on app/src/main/java/com/somestupidappproject/betago/game/Game.java:
```diff
@@ -61,7 +61,15 @@ public boolean undoMostRecentMove() {
+            previousPlayedStone = previousMoves.lastElement().getStone();
+        }
+
+        if (BoardUtils.isValidMove(stone, board, isBlacksMove, previousCapturedStones, previousPlayedStone)) {
```
Seems like it makes more sense to pass in the previous Move instead of splitting up previousCapturedStones and previousPlayedStone

toka13902 committed:
using previous move

sshwang committed:
Ko refactor

sshwang committed:
Resolve Conflicts

# Conflicts:
#	app/src/main/java/com/somestupidappproject/betago/game/Game.java

#	app/src/main/java/com/somestupidappproject/betago/utils/BoardUtils.java

sshwang commented on app/src/main/java/com/somestupidappproject/betago/utils/BoardUtils.java:
```diff
@@ -60,9 +87,60 @@ public static boolean isValidMove(Stone stone, Board board, boolean isBlacksMove
+                HashSet<Stone> stonesNotChecked = new HashSet<Stone>() {{ add(currentStone); }};
+                for (Stone neighbor : neighbors) {
+                    if (!LogicUtil.isAlive(neighbor, board, stonesNotChecked)){
+                        deadStones.add(neighbor);
```
We should return here if deadStones size > 1

esap120 commented on app/src/main/java/com/somestupidappproject/betago/utils/BoardUtils.java:
```diff
@@ -36,12 +37,38 @@
+            ArrayList<Stone> previousCapturedStones,
+            Stone previousStone ) {
         // convert the boolean into an int
         int color = isBlacksMove ? 1 : 2;
```
this looks like an unused variable, plus stone stores the color information already

sshwang committed:
Refactor isKo

Return false earlier if there is more than one dead stone

sshwang committed:
Remove extra argument in isValidMove
